varianteval/core/variant/sv.py

```diff
@@ -1,40 +1,1200 @@
+    sequence = retrieve_instance(tmp_sequence,sequences)
+    load_breakpoint(record,True,tmp_breakpoint)
+    tmp_breakpoint.sequence = sequence
+    tmp_breakpoint.side = BREAKPOINT_SIDE.RIGHT
+    tmp_adjacency.breakpoint1 = retrieve_instance(tmp_breakpoint,breakpoints)
+    load_breakpoint(record,False,tmp_breakpoint)
+    tmp_breakpoint.sequence = sequence
+    tmp_breakpoint.side = BREAKPOINT_SIDE.LEFT
+    tmp_adjacency.breakpoint2 = retrieve(tmp_breakpoint,breakpoints)
+    tmp_event.adjacencies.append(retrieve_instance(tmp_adjacency,adjacencies))
+    retrieve_instance(tmp_event,events)
+
+
+def load_event_ins(record: VariantRecord, contig_lengths: dict[str,int], tmp_event: Event, tmp_adjacency: Adjacency, tmp_breakpoint: Breakpoint, tmp_sequence: Sequence):
+    """
+    Converts ``record`` into an insertion event and stores it in global 
+    dictionary ``events``.
+    """
+    tmp_event.clear()
+    tmp_event.event_type = SV_TYPE.INS
+    
+    load_sequence(record,True,contig_lengths,tmp_sequence)
+    sequence1 = retrieve_instance(tmp_sequence,sequences)
+    load_breakpoint(record,True,tmp_breakpoint)
+    tmp_breakpoint.sequence = sequence1
+    tmp_breakpoint.side = BREAKPOINT_SIDE.RIGHT
+    breakpoint1 = retrieve_instance(tmp_breakpoint,breakpoints)
+    load_breakpoint(record,False,tmp_breakpoint)
+    tmp_breakpoint.sequence = sequence1
+    tmp_breakpoint.side = BREAKPOINT_SIDE.LEFT
+    breakpoint4 = retrieve_instance(tmp_breakpoint,breakpoints)
+    
+    load_sequence(record,False,contig_lengths,tmp_sequence)    
+    sequence2 = retrieve_instance(tmp_sequence,sequences)
+    set_precise_beakpoint(tmp_breakpoint,sequence2,0,BREAKPOINT_SIDE.LEFT)
+    breakpoint2 = retrieve_instance(tmp_breakpoint,breakpoints)
+    set_precise_beakpoint(tmp_breakpoint,sequence2,sequence2.length-1,BREAKPOINT_SIDE.RIGHT)
+    breakpoint3 = retrieve_instance(tmp_breakpoint,breakpoints)
+    
+    tmp_adjacency.breakpoint1 = breakpoint1
+    tmp_adjacency.breakpoint2 = breakpoint2
+    tmp_event.adjacencies.append(retrieve_instance(tmp_adjacency,adjacencies))
+    tmp_adjacency.breakpoint1 = breakpoint3
+    tmp_adjacency.breakpoint2 = breakpoint4
+    tmp_event.adjacencies.append(retrieve_instance(tmp_adjacency,adjacencies))
+    retrieve_instance(tmp_event,events)
+
+
+def load_event_dup(record: VariantRecord, contig_lengths: dict[str,int], tmp_event: Event, tmp_adjacency: Adjacency, tmp_breakpoint: Breakpoint, tmp_sequence: Sequence):
+    """
+    Converts ``record`` into a tandem duplication event and stores it in global
+    dictionary ``events``. 
+    
+    Remark: we assume that a DUP record means that the new adjacency between the
+    end and the beginning of the tandem unit is observed by the caller. This
+    should be implemented better by using field SVCLAIM from the VCF 4.4 spec.
+    """
+    tmp_event.clear()
+    tmp_event.event_type = SV_TYPE.INS
+    
+    load_sequence(record,True,contig_lengths,tmp_sequence)
```

Comment:
looks like there is a lot of redundant code in these loaders -- can we break this up into some helper functions for shared functionality?

Reply:
It _looks_ like that since every variant has to create breakpoints, adjacencies, and events. The details, though, differ a bit for each variant. I vote for keeping the code as it is (they are short procedures anyway).